blocks_extras/extensions/synchronization.py

```diff
@@ -0,0 +1,190 @@
+"""Synchronization of parameters of concurrent training processes.
+
+Use this module to run several Blocks training processes
+concurrently, e.g. to perform Asynchronous SGD [DOWNPOUR_].
+It is built on top of Platoon, a simple framework that makes it easy
+to share parameters and perform communication among multiple training
+jobs. Below is the list of the key concepts introduced in Platoon:
+
+- the training process is called *worker*
+- a separate process called *controller* is sending/receiving messages
+  from/to workers and is responsible for synchronizing their activities
+- each process has its *local parameters*, which are usual Theano shared
+  variables
+- all processes share *global* parameters, which are stored in
+  shared memory (not to be confused with shared Theano variables!)
```

Comment:
I would specify CPU shared memory. The GPU have a shared memory, but it is a different concept.